pinecone/index_connection.go

```diff
@@ -233,17 +730,113 @@ func (idx *IndexConnection) UpdateVector(ctx context.Context, in *UpdateVectorRe
 	return err
 }
 
+// DescribeIndexStatsResponse holds the parameters for the DescribeIndexStatsResponse object,
+// which is returned by the DescribeIndexStats method.
+//
+// Fields:
+//  - Dimension: The dimension of the index.
+//  - IndexFullness: The fullness level of the index. Note: only available on pods-based indexes.
```

Comment:
nit: Does "pod-based" feel better than "pods-based"? Might just be me. 

Also I didn't know this was the case, is it just always 0 for serverless?

Reply:
I like pods-based b/c it's more technically correct but 🤷 .

And shoot... thank you so much for this catch. `DescribeIndexStats` is for both types of indexes, while `DescribeIndexStatsFiltered` is only for pods-based indexes.

Reply:
If you try to run `DescribeIndexStatsFiltered` on a Serverless index, you get this error: 
```
Failed to describe index "serverless-test". Error: rpc error: code = InvalidArgument desc = Serverless and Starter indexes do not support describing index stats with metadata filtering
```